bedrock/newsletter/tests/test_views.py

```diff
@@ -253,33 +249,84 @@ def test_returns_failure__missing_privacy(self, basket_mock):
+        self.assertIn("newsletters", resp.json())
 
 
-@patch("bedrock.newsletter.views.l10n_utils.render", return_value=HttpResponse())
 class TestNewsletterStringsJson(TestCase):
-    def test_newsletter_strings_json(self, render_mock):
-        req = RequestFactory().get("/newsletter/newsletter-strings.json")
-        req.locale = "en-US"
-        newsletter_strings_json(req)
-        template = render_mock.call_args[0][1]
-        self.assertTrue(template == "newsletter/includes/newsletter-strings.json")
+    def test_newsletter_strings_json(self):
+        resp = self.client.get(reverse("newsletter.strings"))
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "newsletter/includes/newsletter-strings.json")
+
+
+class TestNewsletterExistingPage(TestCase):
+    def setUp(self):
+        self.token = str(uuid.uuid4())
+        self.url = reverse("newsletter.existing", kwargs={"token": self.token})
+        self.url_no_token = reverse("newsletter.existing.no-token")
+
+    def test_with_token(self):
+        resp = self.client.get(self.url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "newsletter/management.html")
+
+    def test_template_context(self):
+        resp = self.client.get(self.url)
+        context = resp.context
+        self.assertEqual(context["source_url"], reverse("newsletter.existing.no-token"))
+        self.assertEqual(context["did_confirm"], False)
+
+    def test_no_token_okay(self):
+        resp = self.client.get(self.url_no_token)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "newsletter/management.html")
+
+
+class TestNewsletterCountryPage(TestCase):
+    def setUp(self):
+        self.token = str(uuid.uuid4())
+        self.url = reverse("newsletter.country", kwargs={"token": self.token})
+        self.url_no_token = reverse("newsletter.country.no-token")
+
+    def test_with_token(self):
+        resp = self.client.get(self.url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "newsletter/country.html")
+
+    def test_template_context(self):
+        resp = self.client.get(self.url)
+        context = resp.context
+        self.assertEqual(context["action"], f"{settings.BASKET_URL}/news/user-meta/")
+        self.assertEqual(context["recovery_url"], reverse("newsletter.recovery"))
+
+    def test_no_token_okay(self):
+        resp = self.client.get(self.url_no_token)
+        self.assertEqual(resp.status_code, 200)
+        self.assertTemplateUsed(resp, "newsletter/country.html")
+
+    def test_country_form(self):
+        resp = self.client.get(self.url + "?geo=fr")
+        form = resp.context["form"]
+        self.assertEqual(form.initial, {"country": "fr"})
+        country_choices = dict(form.fields["country"].choices)
+        # We always set the locale to 'en-US' in the form, thus why this isn't "États-Unis".
```

Comment:
Just calling this out here. In the form we have the ability to pass in the request locale so these are translated in the form. But we're not. I'm curious why not? But this at least adds a test for that.

Reply:
It looks like the page is en-US only from what I can tell. Not sure why that is though tbh